compiler/rustc_session/src/config.rs

```diff
@@ -1418,102 +1440,83 @@ impl CheckCfg {
+        // Though symbols inserted conditionally in `default_configuration` are
+        // inserted unconditionally here.
 
-        let atomic_values = &[
-            sym::ptr,
-            sym::integer(8usize),
-            sym::integer(16usize),
-            sym::integer(32usize),
-            sym::integer(64usize),
-            sym::integer(128usize),
-        ];
+        ins!(sym::debug_assertions, no_values);
+
+        // njn: not in default_configuration!
+        ins!(sym::doc, no_values);
+        // njn: not in default_configuration!
+        ins!(sym::doctest, no_values);
+
+        // njn: not in default_configuration!
+        ins!(sym::miri, no_values);
+
+        ins!(sym::overflow_checks, no_values);
+
+        ins!(sym::panic, empty_values).extend(&PanicStrategy::all());
+
+        ins!(sym::proc_macro, no_values);
+
+        ins!(sym::relocation_model, empty_values).extend(RelocModel::all());
 
         let sanitize_values = SanitizerSet::all()
             .into_iter()
             .map(|sanitizer| Symbol::intern(sanitizer.as_str().unwrap()));
+        ins!(sym::sanitize, empty_values).extend(sanitize_values);
 
-        let relocation_model_values = RelocModel::all();
-
-        // Unknown possible values:
-        //  - `target_feature`
-        for name in [sym::target_feature] {
-            self.expecteds.entry(name).or_insert(ExpectedValues::Any);
-        }
-
-        // No-values
-        for name in [
-            sym::doc,
-            sym::miri,
-            sym::unix,
-            sym::test,
-            sym::doctest,
-            sym::windows,
-            sym::proc_macro,
-            sym::debug_assertions,
-            sym::overflow_checks,
-            sym::target_thread_local,
-        ] {
-            self.expecteds.entry(name).or_insert_with(no_values);
-        }
-
-        // Pre-defined values
-        self.expecteds.entry(sym::panic).or_insert_with(empty_values).extend(panic_values);
-        self.expecteds.entry(sym::sanitize).or_insert_with(empty_values).extend(sanitize_values);
-        self.expecteds
-            .entry(sym::target_has_atomic)
-            .or_insert_with(no_values)
-            .extend(atomic_values);
-        self.expecteds
-            .entry(sym::target_has_atomic_load_store)
-            .or_insert_with(no_values)
-            .extend(atomic_values);
-        self.expecteds
-            .entry(sym::target_has_atomic_equal_alignment)
-            .or_insert_with(no_values)
-            .extend(atomic_values);
-        self.expecteds
-            .entry(sym::relocation_model)
-            .or_insert_with(empty_values)
-            .extend(relocation_model_values);
-
-        // Target specific values
+        // njn: sym::sanitizer_cfi_normalize_integers missing!
+        // njn: sym::sanitizer_cfi_normalize_pointers missing!
+
+        // Unknown possible values
+        // njn: not in default_configuration!
+        ins!(sym::target_feature, || ExpectedValues::Any);
```

Comment:
These are set at https://github.com/rust-lang/rust/blob/9bf30ebdfa50db6286d9400615c2bcb324c1a587/compiler/rustc_interface/src/util.rs#L43 based on the list returned by the codegen backend. rustc_codegen_ssa has a list of known target features and their stability, but we should allow any target feature as a new target or rustc version may introduce new target features.

Reply:
As set above by `@bjorn3` it's not set in `default_configuration` but after in the pipeline in the `rustc_interface` crate, so we should at least not lint on the name.

As for linting on the values, I didn't added them because I didn't think we have a way to get all the possibles values for all the possibles targets, but if we have a complete list in `rustc_codegen_ssa` we could (should?) use it.

Reply:
:+1: